crates/katana/storage/provider/src/providers/fork/backend.rs

```diff
@@ -275,11 +328,26 @@ where
 
             // poll all pending requests
             for n in (0..pin.pending_requests.len()).rev() {
-                let mut fut = pin.pending_requests.swap_remove(n);
+                let (fut_key, mut fut) = pin.pending_requests.swap_remove(n);
                 // poll the future and if the future is still pending, push it back to the
                 // pending requests so that it will be polled again
-                if fut.poll_unpin(cx).is_pending() {
-                    pin.pending_requests.push(fut);
+                match fut.poll_unpin(cx) {
+                    Poll::Pending => {
+                        pin.pending_requests.push((fut_key, fut));
+                    }
+                    Poll::Ready(res) => {
+                        let sender_vec = pin
+                            .request_dedup_map
+                            .get(&fut_key)
+                            .expect("failed to get sender vector");
+
+                        // Send the response to all the senders waiting on the same request
+                        sender_vec.iter().for_each(|sender| {
+                            sender.send(res.clone()).unwrap_or_else(|_| error!(target: LOG_TARGET, "failed to send result of request {:?} to sender {:?}", fut_key, sender));
+                        });
```

Comment:
let's use structured fields 

```suggestion
                            sender.send(res.clone()).unwrap_or_else(|error| {
                            	error!(target: LOG_TARGET, key = ?fut_key, %error, "Failed to send result.")
                            });
```